### ISSUE TITLE
Compiler warning in Bio/trie.c signed and unsigned int

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1043]
eg https://travis-ci.org/biopython/biopython/jobs/193783500 says

```
Bio/trie.c: In function ‘Trie_set’:
Bio/trie.c:155:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
Bio/trie.c: In function ‘_get_approximate_trie’:
Bio/trie.c:381:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
Bio/trie.c:393:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
Bio/trie.c: In function ‘_iterate_helper’:
Bio/trie.c:496:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

**Comment by peterjc** [comment]
Possible fix on https://github.com/biopython/biopython/pull/1047

**Comment by peterjc** [comment]
Fixed via #1047
